### PR TITLE
More Visual Refresh Tweaks

### PR DESCRIPTION
**Pull request by emplums**
This PR makes the tweaks listed out in #731 ✨

- [x] Update selected style for primary and outline box shadow to 2px instead of 1px
- [x] Update UnderlineNav & SideNav states (removed bar on the left on hover/focus)
- [x] SideNav should have had a white background, but it was grey so I updated that
- [x] The grey background hover on SideNav items was overlapping the border radius of the container, so I added border radius to the first and last SideNav items
- [x] Fix SelectMenu tab styles
- [x] disabled state for buttons with color icon
- [x] Change back the text in labels and topic tags from font-weight 500 to 400
- [x] Add label colors to the theme
- [x] Spacing between select menu header and select menu tabs was mistakingly set as 24px. I have no changed it to 16px. 
- [x] Update blue text in Links, Outline button to blue[5]
- [x] A bunch of button tweaks: https://github.com/github/design-systems/issues/770#issuecomment-605783773 

#### Notes
- I added the label colors to the theme so that folks can pull from the theme if they'd like to use them, but we might want to think about whether or not we want to add some kind of `colorScheme` prop in the future to make using these easier.

- I removed `primer-colors` and `primer-typography` and moved over to `@primer/primitives`

- I had one question for @auareyou about the active/selected/hover state for SideNav that I left in the comments

### Merge checklist
- [ ] Added or updated TypeScript definitions (`index.d.ts`) if necessary
- [ ] Added/updated tests
- [ ] Added/updated documentation
- [ ] Tested in Chrome
- [ ] Tested in Firefox
- [ ] Tested in Safari
- [ ] Tested in Edge

Take a look at the [What we look for in reviews](https://github.com/primer/components/blob/master/CONTRIBUTING.md#what-we-look-for-in-reviews) section of the contributing guidelines for more information on how we review PRs.

### CODE REVIEW

File: src/SideNav.js [hunk 1]
```diff
@@ -95,26 +105,16 @@ SideNav.Link = styled(Link).attrs(props => {
...
       outline: none;
-
-      // Bar on the left
-      &::before {
-        background-color: ${get('colors.gray.4')};
-      }
-    }
-
-    &:active {
-      background-color: ${get('colors.white')};
```
Comment (emplums) [inline src/SideNav.js:106]: @auareyou in the Figma sticker sheet, for things like buttons it looks like you use the term `selected` to mean `active` (AKA the mouse is currently clicking on the item). 

For the [vertical navigation](https://www.figma.com/file/k2BopyqPZy5Wz7hAiZbnPi/Component-refresh?node-id=4031%3A228) mockup, are the `selected` state styles meant for when item in the menu is currently selected (the user had clicked on that link and is currently on that page), or does that mean the item is `active` (the mouse is currently pressed down and clicking on the item)?
image

 I interpreted the `selected` state in the design as meaning that the item in the menu is currently selected... but should we have a special state for `active`? Right now I'm just using the hover state for active (gray background):
image

File: src/ButtonStyles.js [hunk 2]
```diff
@@ -27,4 +27,8 @@ export default css`
   &:disabled {
     cursor: default;
   }
+
+  &[disabled] svg {
```
Comment (colebemis) [inline src/ButtonStyles.js:31]: Just curious, why did you use `[disabled]` instead of `:disabled`?

File: src/Label.js [hunk 3]
```diff
@@ -43,7 +43,7 @@ const sizeVariant = variant({
 
 const Label = styled('span')`
   display: inline-block;
-  font-weight: 600;
+  font-weight: 400;
```
Comment (colebemis) [inline src/Label.js:46]: Should we use a theme value here?

```suggestion
  font-weight: ${get('fontWeights.normal')};
```

File: src/theme.js [hunk 4]
```diff
@@ -1,6 +1,7 @@
-import {black, white, gray, blue, green, orange, purple, red, yellow} from 'primer-colors'
-import {lineHeights} from 'primer-typography'
+import {colors as colorPrimitives, typography} from '@primer/primitives'
 
+const {lineHeights} = typography
+const {black, white, pink, gray, blue, green, orange, purple, red, yellow} = colorPrimitives
```
Comment (BinaryMuse) [inline src/theme.js:4]: 💯

File: src/ButtonStyles.js (hunk 2 above)
Comment (emplums) [reply src/ButtonStyles.js:31]: I'm actually not sure if there's a difference, I saw `[disabled]` in an example @simurai gave when we were chatting about it and it worked so I didn't think much about it 🤔

Comment (emplums) [reply src/ButtonStyles.js:31]: There's some context here if you're interested: https://stackoverflow.com/questions/20141450/should-i-use-css-disabled-pseudo-class-or-disabled-attribute-selector-or-is-i

I don't think we support IE 11 but I'll change it back to `:disabled` just for consistency sake

**Comment by emplums** [comment]
Going to release this without the Avatar changes as @colinkeany mentioned it will be a bit more complicated than just adjusting the border radius across the board. We can do this in a different PR/release.

File: src/SideNav.js (hunk 1 above)
Comment (auareyou) [reply src/SideNav.js:106]: I think your interpretation is correct. I was using selected and active as the same state without realizing selected means a different thing in components like this. 👍

**Comment by emplums** [comment]
@auareyou regarding Simon's comment here: https://github.com/github/design-systems/issues/770#issuecomment-597369833

Does this mean that all accents should be updated to `f9826c` or just the border only in `menu`, `UnderlineNav` and `SideNav`?

I don't think these are stickersheet yet so wasn't sure if I should implement it in Primer Components. If so - what is the coral background used for? Is it the hover state?

**Comment by emplums** [comment]
Going to do the Flash updates in a separate branch as well, as we should probably do an API refactor of Flash to use standard icons for success/failure/warn

**Pull request merged**